src/core_modules/capture-core/components/Pages/Enrollment/MissingMessage.component.js

```diff
@@ -0,0 +1,154 @@
+// @flow
+import React, { useEffect, useState } from 'react';
+import i18n from '@dhis2/d2-i18n';
+import { useHistory } from 'react-router';
+import { useDispatch, useSelector } from 'react-redux';
+import { useScopeInfo } from '../../../hooks/useScopeInfo';
+import { useMissingCategoriesInProgramSelection } from '../../../hooks/useMissingCategoriesInProgramSelection';
+import { scopeTypes } from '../../../metaData/helpers/constants';
+import { urlArguments } from '../../../utils/url';
+import { IncompleteSelectionsMessage } from '../../IncompleteSelectionsMessage';
+import LinkButton from '../../Buttons/LinkButton.component';
+import { useEnrollmentInfo } from './hooks';
+
+export const missingStatuses = {
+    TRACKER_PROGRAM_WITH_ZERO_ENROLLMENTS_SELECTED: 'TRACKER_PROGRAM_WITH_ZERO_ENROLLMENTS_SELECTED',
+    EVENT_PROGRAM_SELECTED: 'EVENT_PROGRAM_SELECTED',
+    MISSING_ENROLLMENT_SELECTION: 'MISSING_ENROLLMENT_SELECTION',
+    MISSING_PROGRAM_CATEGORIES_SELECTION: 'MISSING_PROGRAM_CATEGORIES_SELECTION',
+    MISSING_PROGRAM_SELECTION: 'MISSING_PROGRAM_SELECTION',
+};
+
+const useMissingStatus = () => {
+    const dispatch = useDispatch();
+    const [missingStatus, setStatus] = useState(null);
+
+    const { programId, enrollmentId } =
+      useSelector(({ router: { location: { query } } }) =>
+          ({
+              teiId: query.teidId,
+              programId: query.programId,
+              enrollmentId: query.enrollmentId,
+          }),
+      );
+
+    const { scopeType } = useScopeInfo(programId);
+    const { programSelectionIsIncomplete } = useMissingCategoriesInProgramSelection();
+    const { programHasEnrollments, enrollmentsOnProgramContainEnrollmentId } = useEnrollmentInfo(enrollmentId, programId);
+    useEffect(() => {
```

Comment:
I pretty sure we can simplify this and use `useMemo` instead? This will consequently make `setStatus` unnecessary.

Reply:
Great thinking, thanks!